### PR TITLE
Never return 0 on failure

### PR DESCRIPTION
**Pull request by JasperCraeghs**
Since tag 0.8.0 (https://github.com/melexis/warnings-plugin/commit/a32f4ebbccb811de9001af8a7f642d2cac96b228) there has been an unwanted behavior of the plugin where it can return 0 on failure. 
Let's return 1 instead of 0 if the number of warnings is 0, but we expected more than 0 warnings. 
For example, we don't want the plugin to return 0 when there are 0 warnings counted and we use `--minwarnings 1`.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -234,7 +234,7 @@ command:
 Parse for XMLRunner errors
 --------------------------
 
-When you run [XMLRunner](https://github.com/xmlrunner/unittest-xml-reporting)
+When you run XMLRunner_,
 the errors are reported on the output, but they are not marked as failures in
 the test reports xml files. Since command exits as 1, we could not detect tests
```
Comment (JasperCraeghs) [inline README.rst]: > Since command exits as 1

Should this be updated?

Comment (Letme) [reply README.rst]: No, xmlrunner command still exits as 1 and that is why we need warnings plugin...

**Comment by JasperCraeghs** [comment]
Ready for merge

File: src/mlx/warnings_checker.py [hunk 2]
```diff
@@ -124,22 +124,27 @@ def return_check_limits(self):
...
-            print("Number of warnings ({0.count}) is lower than the minimum limit ({0.warn_min}). "
-                  "Returning error code 1.".format(self))
-            return self.count
+            error_reason = "lower than the minimum limit"
         else:
             print("Number of warnings ({0.count}) is between limits {0.warn_min} and {0.warn_max}. Well done."
                   .format(self))
-        return 0
+            return 0
+        # code below is only reached after failure; don't allow error code 0
```
Comment (Letme) [inline src/mlx/warnings_checker.py]: This is an ugly solution, but at this moment can't think of more elegant.

Comment (Letme) [reply src/mlx/warnings_checker.py]: Goto does not exist but functions do. Maybe we put below printout to function and then call return with function call which would return proper error value?

Comment (JasperCraeghs) [reply src/mlx/warnings_checker.py]: I agree that it was ugly. I added a function as you suggested: 1810e3bb21f0cbcc6a0427f409c7d3c87c8fee03

**Pull request merged**